src/main/java/com/redhat/devtools/lsp4ij/features/foldingRange/LSPFoldingRangeBuilder.java

```diff
@@ -62,8 +63,8 @@ public class LSPFoldingRangeBuilder extends CustomFoldingBuilder {
     protected void buildLanguageFoldRegions(@NotNull List<FoldingDescriptor> descriptors,
                                             @NotNull PsiElement root,
                                             @NotNull Document document, boolean quick) {
-        // if quick flag is set, we do nothing here
-        if (quick) {
+        // if quick flag is set and not testing, we do nothing here
+        if (quick && !ApplicationManager.getApplication().isUnitTestMode()) {

```

Comment:
Tests are run in dumb mode so if we want to test this, we need to allow that, at least for test execution.